client/src/components/CheckoutOptions.jsx

```diff
@@ -1,20 +1,69 @@
+/* eslint-disable arrow-body-style */
+/* eslint-disable no-useless-constructor */
+/* eslint-disable class-methods-use-this */
 import React from 'react';
 
 
 class CheckoutOptions extends React.Component {
   constructor(props) {
     super(props);
+
+    this.state = {
+      colorModifier: 1,
+      quantity: 1,
+    };
+
+    this.handleColorChoice = this.handleColorChoice.bind(this);
+    this.handleQuantity = this.handleQuantity.bind(this);
   }
 
+  handleColorChoice(e) {
+    this.setState({
+      colorModifier: Number(e.target.value),
+    }, () => {
+      this.props.handleColorChoice(this.state.colorModifier);
+    });
+  }
 
+  handleQuantity(e) {
+    this.setState({
+      quantity: Number(e.target.value),
+    }, () => {
+      this.props.handleQuantity(this.state.quantity);
+    });
+  }
 
   render() {
+    const quantityOptions = [];
+    for (let i = 2; i < 21; i += 1) {
+      quantityOptions.push(i);
+    }
+
     return (
-      <div></div>
-    )
+      <div className="checkout-options-container">
+          <label className="select-label">Color</label>
+          <select id="select-color" onChange={this.handleColorChoice}>
+            <option value="" disabled selected>Select a Color</option>
+            {this.props.colors.map((color) => {
+              return (
+                <option value={color.price_modifier}>{color.color_name}</option>
+              );
+            })}
```

Comment:
if you wanted, you could use implicit returnning of single-line arrow functions here
```
{this.props.colors.map((color) => (
  <option value={color.price_modifier}>{color.color_name}</option>
))}
```